code/modules/mob/living/status_procs.dm

```diff
@@ -118,311 +103,527 @@
-	return SetEyeBlurry(new_value, updating)
+/mob/living/proc/AdjustEyeBlurry(amount, bound_lower = 0, bound_upper = INFINITY)
+	SetEyeBlurry(clamp(amount, bound_lower, bound_upper))
 
 // HALLUCINATION
+/mob/living/proc/AmountHallucinate()
+	RETURN_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_HALLUCINATION)
 
-/mob/living/Hallucinate(amount)
-	SetHallucinate(max(hallucination, amount))
+/mob/living/proc/Hallucinate(amount)
+	SetHallucinate(max(AmountHallucinate(), amount))
 
-/mob/living/SetHallucinate(amount)
-	hallucination = max(amount, 0)
+/mob/living/proc/SetHallucinate(amount)
+	SET_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_HALLUCINATION, amount)
 
-/mob/living/AdjustHallucinate(amount, bound_lower = 0, bound_upper = INFINITY)
-	var/new_value = directional_bounded_sum(hallucination, amount, bound_lower, bound_upper)
-	SetHallucinate(new_value)
+/mob/living/proc/AdjustHallucinate(amount, bound_lower = 0, bound_upper = INFINITY)
+	SetHallucinate(clamp(amount + AmountHallucinate(), bound_lower, bound_upper))
 
 // JITTER
+/mob/living/proc/AmountJitter()
+	RETURN_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_JITTER)
 
-/mob/living/Jitter(amount, force = 0)
-	SetJitter(max(jitteriness, amount), force)
+/mob/living/proc/Jitter(amount, ignore_canstun = FALSE)
+	SetJitter(max(AmountJitter(), amount), ignore_canstun)
 
-/mob/living/SetJitter(amount, force = 0)
+/mob/living/proc/SetJitter(amount, ignore_canstun = FALSE)
 	// Jitter is also associated with stun
-	if(status_flags & CANSTUN || force)
-		jitteriness = max(amount, 0)
+	if(IS_STUN_IMMUNE(src, ignore_canstun))
+		return
+	SET_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_JITTER, amount)
 
-/mob/living/AdjustJitter(amount, bound_lower = 0, bound_upper = INFINITY, force = 0)
-	var/new_value = directional_bounded_sum(jitteriness, amount, bound_lower, bound_upper)
+/mob/living/proc/AdjustJitter(amount, bound_lower = 0, bound_upper = INFINITY, force = 0)
+	var/new_value = directional_bounded_sum(AmountJitter(), amount, bound_lower, bound_upper)
 	SetJitter(new_value, force)
 
 // LOSE_BREATH
 
-/mob/living/LoseBreath(amount)
-	SetLoseBreath(max(losebreath, amount))
+/mob/living/proc/AmountLoseBreath()
+	RETURN_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_LOSE_BREATH)
 
-/mob/living/SetLoseBreath(amount)
+/mob/living/proc/LoseBreath(amount)
+	SetLoseBreath(max(AmountLoseBreath(), amount))
+
+/mob/living/proc/SetLoseBreath(amount)
 	if(HAS_TRAIT(src, TRAIT_NOBREATH))
-		losebreath = 0
-		return FALSE
-	losebreath = max(amount, 0)
+		return
+	SET_STATUS_EFFECT_STRENGTH(STATUS_EFFECT_LOSE_BREATH, amount)
 
-/mob/living/AdjustLoseBreath(amount, bound_lower = 0, bound_upper = INFINITY)
-	var/new_value = directional_bounded_sum(losebreath, amount, bound_lower, bound_upper)
-	SetLoseBreath(new_value)
+/mob/living/proc/AdjustLoseBreath(amount, bound_lower = 0, bound_upper = INFINITY)
+	SetLoseBreath(clamp(amount + AmountLoseBreath(), bound_lower, bound_upper))
```

Comment:
Replacing `directional_bounded_sum` with `clamp` is semantically wrong on all of those procs, and leads to bugs such as Epi giving losebreath unconditionally.

As i see it you have three options here:
- leave the clamp in, and fix at the call sites all the arising bugs from code calling the `AdjustX` with bounds specified - the way you did it for Epi. But that's lots of work, and doesn't prevent new bugs from being introduced if someone forgets to add an `if`.
Also the `clamp` doesn't do anything really, since it's clamped to `0` in `SetX` anyway, and infinity is infintity.
- change it back to `directional_bounded_sum`. Arguably that's worse performance than the cheap `clamp` but that's the price we pay for correctness.
- if perf is really important, then make two sets of the Adjust procs:
```dm
/mob/proc/AdjustX(value)  // no bounds
  SetX(X, X+value) // also not even a clamp

/mob/proc/AdjustXBounded(value, bound_lower=0, bound_upper=INFINITY)  // yes bounds
  var/new_value = directional_bounded_sum(X, value, bound_lower, bound_upper)
  SetX(new_value)
```

Reply:
moved clamp to directional bounded sum again.